unison-core/src/Unison/ABT.hs

```diff
@@ -528,14 +528,13 @@ find' :: (Ord v, Foldable f, Functor f)
 find' p = Unison.ABT.find (\t -> if p t then Found t else Continue)
 
 instance (Foldable f, Functor f, Eq1 f, Var v) => Eq (Term f v a) where
-  -- alpha equivalence, works by renaming any aligned Abs ctors to use a common fresh variable
+  -- alpha equivalence, works by renaming any aligned Abs ctors to use a common variable
   t1 == t2 = go (out t1) (out t2) where
     go (Var v) (Var v2) | v == v2 = True
     go (Cycle t1) (Cycle t2) = t1 == t2
     go (Abs v1 body1) (Abs v2 body2) =
       if v1 == v2 then body1 == body2
-      else let v3 = freshInBoth body1 body2 v1
-           in rename v1 v3 body1 == rename v2 v3 body2
+      else rename v1 v2 body1 == body2
```

Comment:
@dolio I'm not sure why the old code felt the need to rename both variables, this seems fine to just rename one to  match the other, right?

Reply:
I think it's possible that `body1` has free occurrences of `v2` that will cause erroneous results. Like:

    λ v1 -> f v1 v2
    λ v2 -> f v2 v2

Reply:
Good catch, glad I got your review. I’ll revert and do a proper fix of rename I think. Did you have a failing test of it somewhere?

Might also do the de Bruijn based comparison, using annotateBound.